File: RandomMatrixGeneratorPerl/randommatrixgenerator.pl
#!/usr/bin/perl

use strict;
use warnings;


################################################################################
# Created by: Jordan Hartwick
# Date: May 29, 2016 @ 11:24 PM
# Usage: perl randommatrixgenerator.pl
#
# Purpose: Prints a matrix with a specified amount of rows and columns where all 
# values are less than the max argument.
#
# Test Run
################################################################################
# perl randommatrixgenerator.pl
# Type first and second dimensions: 2 3
# Type a max value: 10
# [9,5,4]
# [6,4,7]
################################################################################

# The following comment would create an array if uncommented.
# my @matrix

# Get the dimensions of the matrix and put them into an array.
print "Type first and second dimensions: ";
my $dim = <STDIN>;
my @dims = split(/ /, $dim);

# Get the max value that can be generated.
print "Type a max value: ";
my $max = <STDIN>;

# Print out the random numbers in the form of a matrix.
for(my $i=0; $i < $dims[0]; $i++) {

	print "[";
	for(my $j=0; $j < $dims[1]; $j++) {
	
		# The following comment would store a random value in the matrix.
		# $matrix[$i][$j] = int rand($max);
		if($j == $dims[1]-1) {
			print int rand($max);
		} else {
			printf("%d,",int rand($max));
		}
	}
	print "]\n";
}

File: RandomMatrixGenerator/RandomMatrixGenerator.java
import java.util.Arrays;
import java.util.Random;


/*******************************************************************************
 * Created by: Jordan Hartwick
 * Date: May 29, 2016 @ 9:50 PM
 * Usage: java RandomMatrixGenerator <rows> <columns> <maxvalue>
 *
 * Purpose: Generates a random matrix with a specified amount of rows and columns
 * where all values are less than the max value.
 *
 * Test Run:
 *******************************************************************************
 * java RandomMatrixGenerator 2 3 6
 * [0, 1, 4]
 * [1, 0, 1]
 */
public class RandomMatrixGenerator {
	
	public static void main(String[] args) {
		Random rand = new Random();
		if(args == null || args[0] == null || args[1] == null || args[2] == null) {
			System.out.println("Usage: java RandomMatrixGenerator rows columns maxvalue");
		}
		
		int rows = Integer.parseInt(args[0]), 
			cols = Integer.parseInt(args[1]), 
			maxV = Integer.parseInt(args[2]);
		
		int[][] matrix = new int[rows][cols];
		for(int r = 0; r < rows; r++) {
			for(int c = 0; c < cols; c++) {
				matrix[r][c] = rand.nextInt(maxV);
			}
			System.out.println(Arrays.toString(matrix[r]));
		}
	}
}
